### ISSUE TITLE
Using several optimizers for one output during training 

### ISSUE THREAD

**Issue by carlthome** [fchollet/keras#2534]
I wonder if it would be possible to use several optimizers for an output and pool them with a strategy, such as picking the new weights from the optimizer which will result in the lowest validation loss, or simply averaging all optimizers proposed new weights.

**Comment by braingineer** [comment]
Ensemble optimizers. Haha.

But, in general, there's no reason why not.  [Lasagne applies nesterov](https://github.com/Lasagne/Lasagne/blob/master/lasagne/updates.py#L243) to existing gradients.  

I wrapped `Adam` with a SARSA eligbility trace
```python
class SARSA_Adam(Adam):
    '''SARSA update technique for reinforcement learning applications. 

    # Arguments
    '''
    def __init__(self, trace_decay=0.9, time_decay=0.9, *args, **kwargs):
        super(SARSA_Adam, self).__init__(*args, **kwargs)
        self.e_trace = None
        self.trace_decay = trace_decay
        self.time_decay = time_decay

    def get_gradients(self, loss, params):
        grads = super(SARSA_Adam, self).get_gradients(loss, params)
        out_grads = []
        for p,g in zip(params, grads):
            if p.name == "value_approximation_W":
                # lambda * gamma * e
                self.e_trace = K.variable(np.zeros(K.get_value(p).shape))
                self.e_update = self.trace_decay * self.time_decay * self.e_trace
                # + dV/dw or dQ/dw ; divide by td to recover it from MSE
                self.e_update += g/self.td
                g = self.td*self.e_update
            out_grads.append(g)
        return out_grads

    def set_td(self, td_tensor):
        """ gives the optimizer a handle on the Temporal Difference error

        # Arguments
            td_tensor: keras tensor.  Output from TD calculations. 
        """
        self.td = td_tensor

    def get_updates(self, params, constraints, loss):
        updates = super(SARSA_Adam, self).get_updates(params, constraints, loss)
        if self.e_trace is not None:
            updates.append((self.e_trace, self.e_update))
```

So really, you could subclass `Optimizer` to create `EnsembleOptimizer` or something, which accepts a list of instantiated optimizers.
